### ISSUE TITLE
Collapse - Horionztal animation

### ISSUE THREAD

**Issue by willd-i2o** [react-bootstrap/react-bootstrap#5977]
As of v5.1 Bootstrap now supports horizontal collapse animations with the `.collapse-horizontal` class https://getbootstrap.com/docs/5.1/components/collapse/#horizontal

I'm guessing the Collapse component will need to be updated to use this when the `dimension` prop === `width`? (`<Collapse dimension="width"></Collapse>` )

**Comment by kyletsang** [comment]
Sounds reasonable.  Currently, when computedDimension, it adds a `.width` class though, but I don't see that in the bootstrap docs.  Maybe we can replace that class?  

cc @jquense 

We'd be happy to take a PR to add this feature.

**Comment by jquense** [comment]
yeah that was the placeholder from v4. we should update it to whatever is actually supported

**Comment by nsi319** [comment]
Hi @kyletsang 
Can I work on adding horizontal collapse to Collapse component?

**Comment by kyletsang** [comment]
@nsi319 feel free to take on any of the issues here.

**Comment by nsi319** [comment]
@kyletsang Should there be a new Component which does horizontal animation (similar to `Fade`) or should it come in Collapse when `dimension` prop === width?

**Comment by nsi319** [comment]
Hi @kyletsang. When the `dimension` prop === `height`, the existing Collapse component is sufficient. When the value is `width`, we need to make the updates. Could you guide me on how to introduce horizontal animations to it?

**Comment by kyletsang** [comment]
@nsi319 

We can replace the `.width` class that's added here with `.collapse-horizontal`:
https://github.com/react-bootstrap/react-bootstrap/blob/74894de1b50e005280d0270719a4a36e2bf4026c/src/Collapse.tsx#L248

ie.

```
computedDimension === 'width' && 'collapse-horizontal',
```
